File: Project Phase Camp/2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        neg = []
        pos = []
        res = []
        
        for n in nums:
            if n < 0:
                neg.append(n)
            
            else:
                pos.append(n)
        
        n = 0
        p = 0
        fn = False
        fp = True
        
        while p + n < len(nums):
            if fp:
                res.append(pos[p])
                p += 1
                fp = False
                fn = True
            
            else:
                res.append(neg[n])
                n += 1
                fp = True
                fn = False
        
        return res
        
        

File: Project Phase Camp/2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var rearrangeArray = function(nums) {
    const neg = [];
    const pos = [];
    
    for(let item of nums){
        if(item < 0){
            neg.push(item);
        }else{
            pos.push(item);
        }
    }
    
    let n = 0;
    let p = 0;
    
    let fn = false;
    let fp = true;
    
    let res = [];
    
    while(n + p < nums.length){
        if(fn){
            res.push(neg[n]);
            n += 1;
            fn = false;
            fp = true
        }else{
            res.push(pos[p]);
            p += 1;
            fn = true;
            fp = false;
        }
    }
    
    return res;
};